src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -848,6 +855,8 @@ public int GetHashCode((string, NuGetVersion) library)
 
             private void WriteAnalyzers()
             {
+                HashSet<string>? excludedAnalyzers = NuGetUtils.GetExcludedAnalyzers(_lockFile, _task.ProjectLanguage, _task.CompilerApiVersion);
```

Comment:
I would like to see the logic to check if an analyzer applies refactored.  However, feel free to merge as is since you're crunched for time.

The way this is structured seems confusing.  `GetExcludedAnalyzers` ends up calling `IsApplicableAnalyzer`, but then returns a HashSet which is passed back in to `IsApplicableAnalyzer`.

I think it would be easier to understand (and possibly faster) if `IsApplicableAnalyzer` simply contained all the logic to determine whether an analyzer was applicable, including the multi-targeting logic, instead of separately computing a list of excluded analyzers.  If needed, that could be wrapped in a class which could maintain state to perform expensive operations once, lazily, as necessary.

Reply:
Your idea of wrapping the logic in a class allowed me to find a way to make this work.

It isn't possible to make this decision directly in `IsApplicableAnalyzer` because we need to see all the assets before we know which assets apply and which don't. Creating a class to keep track of the necessary information allowed me to track the potential assets, and then add them at the end.